subsys/net/lib/zstream/zstream.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <sys/types.h>
+
+#include <net/zstream.h>
+
+ssize_t zstream_writeall(struct zstream *stream, const void *buf, size_t size,
+			 size_t *written)
+{
+	const char *p = buf;
+	size_t cur_sz = 0;
+
+	do {
+		ssize_t out_sz = zstream_write(stream, p, size);
+
+		if (out_sz == -1) {
+			if (written) {
+				*written = cur_sz;
+			}
+			return -1;
```

Comment:
I tend to dislike coalescing actual code statement to a } line, line breaks are cheap.

Reply:
Fixed.